src/actions.ml

```diff
@@ -0,0 +1,782 @@
+      <&> GitHub_mutations.post_comment ~bot_info ~id
+            ~message:
+              "This PR was postponed. Please update accordingly the milestone \
+               of any issue that this fixes as this cannot be done \
+               automatically."
+  | _ ->
+      Lwt_io.printf "This was not a request inclusion column: ignoring.\n"
+
+let push_action ~bot_info ~base_ref ~commits_msg =
+  Stdio.printf "Merge and backport commit messages:\n" ;
+  let commit_action commit_msg =
+    if string_match ~regexp:"^Merge PR #\\([0-9]*\\):" commit_msg then
+      let pr_number = Str.matched_group 1 commit_msg |> Int.of_string in
+      Lwt_io.printf "%s\nPR #%d was merged.\n" commit_msg pr_number
+      >>= fun () ->
+      GitHub_queries.get_pull_request_id_and_milestone ~bot_info ~owner:"coq"
+        ~repo:"coq" ~number:pr_number
+      >>= fun pr_info ->
+      match pr_info with
+      | Ok (Some (_, pr_id, {backport_info})) ->
+          backport_info
+          |> Lwt_list.iter_p
+               (fun {backport_to; request_inclusion_column; backported_column}
+               ->
+                 if "refs/heads/" ^ backport_to |> String.equal base_ref then
+                   Lwt_io.printf
+                     "PR was merged into the backportig branch directly.\n"
+                   >>= fun () ->
+                   GitHub_mutations.add_pr_to_column ~pr_id
+                     ~column_id:backported_column ~bot_info
+                 else
+                   Lwt_io.printf "Backporting to %s was requested.\n"
+                     backport_to
+                   >>= fun () ->
+                   GitHub_mutations.add_pr_to_column ~pr_id
+                     ~column_id:request_inclusion_column ~bot_info)
+      | Ok None ->
+          Lwt_io.printf "Did not get any backporting info.\n"
+      | Error err ->
+          Lwt_io.printf "Error: %s\n" err
+    else if string_match ~regexp:"^Backport PR #\\([0-9]*\\):" commit_msg then
+      let pr_number = Str.matched_group 1 commit_msg |> Int.of_string in
+      Lwt_io.printf "%s\nPR #%d was backported.\n" commit_msg pr_number
+      >>= fun () ->
+      GitHub_queries.get_backported_pr_info ~bot_info pr_number base_ref
+      >>= function
+      | Ok (Some ({card_id; column_id} as input)) ->
+          Lwt_io.printf "Moving card %s to column %s.\n" card_id column_id
+          >>= fun () -> GitHub_mutations.mv_card_to_column ~bot_info input
+      | Ok None ->
+          Lwt_io.printf "Could not find backporting info for backported PR.\n"
+      | Error e ->
+          Lwt_io.printf "%s" e
+    else Lwt.return ()
+  in
+  fun () -> Lwt_list.iter_s commit_action commits_msg
+
+let repeat_request request =
+  let rec aux t =
+    request
+    >>= fun body ->
+    if String.is_empty body then Lwt_unix.sleep t >>= fun () -> aux (t *. 2.)
+    else Lwt.return body
+  in
+  aux 2.
+
+type build_failure = Warn | Retry | Ignore
+
+let trace_action ~repo_full_name trace =
```

Comment:
This is used in a commented code in the `job_failure` function, so this should be moved above.